src/main/java/org/scijava/ui/swing/widget/SwingChoiceWidget.java

```diff
@@ -97,8 +100,35 @@ public boolean supports(final WidgetModel model) {
 
 	@Override
 	public void doRefresh() {
-		final Object value = get().getValue();
-		if (value.equals(comboBox.getSelectedItem())) return; // no change
-		comboBox.setSelectedItem(value);
+		final String[] choices = get().getChoices();
+		
+		if (!areListsEqual(choices, comboBoxItems())) {
+			comboBox.removeAllItems();	
+			for (int i=0; i<choices.length; i++)
+				comboBox.addItem(choices[i]);
+		} else {
+			final Object value = get().getValue();
+			if (value.equals(comboBox.getSelectedItem())) return;
+			comboBox.setSelectedItem(value);
+		}
+	}
+	
+	private boolean areListsEqual(String[] list1, String[] list2) {
```

Comment:
This method name is misleading, as we're comparing arrays, not lists.

It can be replaced by `Arrays.equals(array1, array2)`, no?

Reply:
Yes indeed! I can add that change to the PR, but I will wait to see if there are others...

Reply:
Thanks for taking a look!

Reply:
I added the change and it worked fine in my tests.